lib/ansible/modules/network/dcnm/dcnm_vrf.py

```diff
@@ -37,237 +39,1042 @@
+        else:
+            if have['vrfId'] != want['vrfId']:
+                logit("diff_for_create - VRF ID cant be updated to a different value")
+                pass
+                # module.exit_json("Can not update the VRF ID") # need to find a proper way to err
+            elif have['serviceVrfTemplate'] != want['serviceVrfTemplate'] or \
+                    have['source'] != want['source'] or \
+                    have['vrfTemplate'] != want['vrfTemplate'] or \
+                    have['vrfExtensionTemplate'] != want['vrfExtensionTemplate']:
+                create = want
+            else:
+                pass
+
+        return create
+
+
+    def update_create_params(self, create):
+
+        if not create:
+            return create
+
+        v_template = 'Default_VRF_Universal' if not create.get('vrf_template') else create['vrf_template']
+        ve_template = 'Default_VRF_Extension_Universal' if not create.get('vrf_extension_template') \
+            else create['vrf_extension_template']
+        src = None if not create.get('source') else create['source']
+        s_v_template = None if not create.get('service_vrf_template') else create['service_vrf_template']
+
+        create_upd = dict()
+        create_upd.update({'fabric': self.fabric})
+        create_upd.update({'vrfName': create['vrf_name']})
+        create_upd.update({'vrfTemplate': v_template})
+        create_upd.update({'vrfExtensionTemplate': ve_template})
+        create_upd.update({'vrfId': create['vrf_id']})
+        create_upd.update({'serviceVrfTemplate': s_v_template})
+        create_upd.update({'source': src})
+
+        template_conf = dict()
+        template_conf.update({'vrfSegmentId': create['vrf_id']})
+        template_conf.update({'vrfName': create['vrf_name']})
+
+        create_upd.update({'vrfTemplateConfig': json.dumps(template_conf)})
 
-    return False
+        return create_upd
 
 
-def vrf_create_payload(module):
+    def get_have(self):
+        """
 
-    payload = dict()
-    payload['fabric'] = module.params['fabric']
-    payload['vrfName'] = module.params['vrf_name']
-    payload['vrfTemplate'] = module.params['vrf_template']
-    payload['vrfExtensionTemplate'] = module.params['vrf_extension_template']
-    payload['vrfTemplateConfig'] = module.params['vrf_template_config']
-    payload['vrfId'] = module.params['vrf_id']
+        Sufficient error checking??
+        Possiibe inputs from DCNM:
+        1. API failures
+        2. Blank responses - no vrfs
+                           - no attachments
+                           - no deployments
+        3. typecast booleans before checks?
+        """
 
-    json_data = json.dumps(payload)
+        have_create = list()
+        have_attach = list()
+        have_deploy = dict()
 
-    return json_data
+        all_vrfs = ''
 
+        method = 'GET'
+        path = '/rest/top-down/fabrics/{}/vrfs'.format(self.fabric)
+
+        resp = dcnm_send(self.module, method, path)
```

Comment:
I find the use of `resp`, `all_vrfs` and `all_vrf_attach` a bit confusing.  What about the following to provide clarity?

```python
method = 'GET'
path = '/rest/top-down/fabrics/{}/vrfs'.format(self.fabric)
vrf_objects = dcnm_send(self.module, method, path)
```

```python
path = '/rest/top-down/fabrics/{}/vrfs/attachments?vrf-names={}'.format(self.fabric, all_vrfs[:-1])
vrf_attach_objects = dcnm_send(self.module, method, path)
```

Reply:
sure, horrible with naming...